nav2_smac_planner/src/node_hybrid.cpp

```diff
@@ -492,6 +515,126 @@ float NodeHybrid::getObstacleHeuristic(
   return 2.0 * obstacle_heuristic_lookup_table[start_index];
 }
 
+inline float distanceHeuristic2D(
+  const int idx, const int size_x, const int goal_x,
+  const int goal_y)
+{
+  int x = std::abs(idx % size_x - goal_x);
+  int y = std::abs(idx / size_x - goal_y);
+  float & table_value = NodeHybrid::astar_2d_h_table[y * size_x + x];
+  return table_value < 0.0f ? table_value = std::hypotf(x, y) : table_value;
```

Comment:
Just curious - how much does caching this L2 norma speed things up (ballpark)? If you don't have numbers handy that's OK as long as you know it actually does speed things up.

Reply:
Well, this caching is maybe just an artifact from my experiments (presuming there may be more to compute than just `hypotf`) and it's possible that it does not bring a speedup. I will test it and remove if unnecessary/detrimental.

Reply:
OK, its just a very large data structure to have around so if there's no speed up, that would decrease memory usage _alot_.

Reply:
@afrixs did you ultimately find this didn't speed anything up (even when having to reprioritize the queue on occasion?)

Reply:
Yes, I added some time logs and there was no speed up at all.